qiime/core/tests/test_archiver.py

```diff
@@ -125,3 +124,21 @@ def test_save_metadata_file_encoding(self):
                 with io.TextIOWrapper(bytes_fh, newline=None,
                                       encoding='utf-8', errors='strict') as fh:
                     self.assertTrue(fh.read().rstrip('\n'))
+
+    def test_metadata_pprint_yaml(self):
+        archive = self.FakeArchiver()
+        fp = os.path.join(self.root_dir, "archive.qza")
+
+        with zipfile.ZipFile(fp, mode="w",
+                             compression=zipfile.ZIP_DEFLATED,
+                             allowZip64=True) as zf:
+            archive._save_metadata(zf, self.root_dir)
+
+            with zf.open(os.path.join(self.root_dir,
+                                      Archiver._METADATA_FILENAME)) as zip_fh:
+                with io.TextIOWrapper(zip_fh, newline=None,
```

Comment:
Agreed! I guess I could not use CM's, and then have a few extra lines for cleanup. I don't see that as being particularly better, though. Your call!